src/backend/backend/auth/bruker.ts

```diff
@@ -1,20 +1,26 @@
+import type { User } from '../typer';
 import type { TexasClient } from './texas';
 import type { Userinfo } from '../../typer/entraid';
-import type { NextFunction, Request, Response } from 'express';
+import type { Request, Response } from 'express';
 
 import axios from 'axios';
 
-import { logRequest } from '../utils';
-import { utledAccessToken } from './tokenUtils';
 import { appConfig } from '../../config';
 import { retry } from '../../http';
 import { LogLevel } from '../../logging/logging';
+import { logRequest } from '../utils';
+import { utledAccessToken } from './authenticate';
 
-// Hent brukerprofil fra sesjon
-export const hentBrukerprofil = () => {
-    return async (req: Request, res: Response) => {
-        if (!req.session) {
-            throw new Error('Mangler sesjon på kall');
+export const hentBrukerprofil = (texasClient: TexasClient) => {
+    return async (req: Request, res: Response): Promise<void> => {
+        const requestToken = utledAccessToken(req);
+        if (!requestToken) {
+            res.sendStatus(401);
+            return;
+        }
+
+        if (req.session.user == null) {
```

Comment:
Dette blir vel mer riktig?
```suggestion
        if (req.session.user === null) {
```